### PR TITLE
Guard against serializing mismatching chunk coordinate

### PR DESCRIPTION
**Pull request by Spottedleaf**
Should help if something dumb happens

### CODE REVIEW

File: Spigot-Server-Patches/0422-Guard-against-serializing-mismatching-chunk-coordina.patch [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+diff --git a/src/main/java/net/minecraft/server/IChunkLoader.java b/src/main/java/net/minecraft/server/IChunkLoader.java
+index 134c76065..25c8b131f 100644
+--- a/src/main/java/net/minecraft/server/IChunkLoader.java
++++ b/src/main/java/net/minecraft/server/IChunkLoader.java
+@@ -106,6 +106,13 @@ public class IChunkLoader extends RegionFileCache implements AutoCloseable {
+ //
+     public void a(ChunkCoordIntPair chunkcoordintpair, NBTTagCompound nbttagcompound) throws IOException { write(chunkcoordintpair, nbttagcompound); } // Paper OBFHELPER
+     public void write(ChunkCoordIntPair chunkcoordintpair, NBTTagCompound nbttagcompound) throws IOException { // Paper - OBFHELPER - (Switched around for safety)
++        // Paper start
++        if (!chunkcoordintpair.equals(ChunkRegionLoader.getChunkCoordinate(nbttagcompound))) {
```
Comment (electronicboy) [inline Spigot-Server-Patches/0422-Guard-against-serializing-mismatching-chunk-coordina.patch:46]: Justification for throw vs fix? jw

Comment (Spottedleaf) [reply Spigot-Server-Patches/0422-Guard-against-serializing-mismatching-chunk-coordina.patch:46]: if neither are consistent we're taking a coin toss for overriding good data

Comment (electronicboy) [reply Spigot-Server-Patches/0422-Guard-against-serializing-mismatching-chunk-coordina.patch:46]: Good point, I mean, I'm guessing that I'm mostly expecting that this is
1) Not the issue
2) Never going to be seen

But, I've seen odder things...

Comment (Spottedleaf) [reply Spigot-Server-Patches/0422-Guard-against-serializing-mismatching-chunk-coordina.patch:46]: if anything would be caught it'll be plugins

**Review by electronicboy — APPROVED** [review]

**Pull request merged**